spec/recipe_spec.rb

```diff
@@ -2,24 +2,26 @@
 describe ".find_in_api" do
   ret = { "this": "is a test" }
   before(:each) do
-    stub_request(:get, "http://www.example.com").
+    stub_request(:get, %r{https://spoonacular-recipe-food-nutrition-v1\.p\.rapidapi\.com/recipes/mealplans/generate.*}).
       with(  headers: {
       'Accept'=>'*/*',
       'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
-      'Host'=>'www.example.com',
-      'User-Agent'=>'Ruby'
-        }).
+      'Host'=>'spoonacular-recipe-food-nutrition-v1.p.rapidapi.com',
+      'User-Agent'=>'Ruby',
+      'X-RapidAPI-Host'=> 'spoonacular-recipe-food-nutrition-v1.p.rapidapi.com',
+      'X-RapidAPI-Key'=> ENV['SPOONACULAR_API_KEY']
+       }).
       to_return(status: 200, body: JSON.generate(ret), headers: {"Content-Type"=> "application/json"})
   end
   
   it 'should be a 200 OK' do
-    expect(Recipe.find_in_api.code).to eq "200"
+    expect(Recipe.find_in_api(1,1,1).code).to eq "200"
   end
   it 'should return a string' do
-    expect(Recipe.find_in_api.body).to be_an_instance_of(String)
+    expect(Recipe.find_in_api(1,1,1).body).to be_an_instance_of(String)
   end
   it 'should include "is a test" in body' do
-    expect(Recipe.find_in_api.body).to include('is a test')
+    expect(Recipe.find_in_api(1,1,1).body).to include('is a test')
```

Comment:
is there a way we can properly mock the arguments instead of just specifying a dummy value to fill the args?